test/engine/SpatialJoinAlgorithmsTest.cpp

```diff
@@ -780,8 +781,277 @@ TEST_P(SpatialJoinParamTest, computeResultSmallDatasetDifferentSizeChildren) {
+  auto checkOutside = [&](const point& point1, const point& startPoint,
+                          const std::vector<box>& bbox,
+                          SpatialJoinAlgorithms* spatialJoinAlg) {
+    // check if the point is contained in any bounding box
+    bool within = spatialJoinAlg->OnlyForTestingWrapperContainedInBoundingBoxes(
+        bbox, point1);
+    if (!within) {
+      GeoPoint geo1{point1.get<1>(), point1.get<0>()};
+      GeoPoint geo2{startPoint.get<1>(), startPoint.get<0>()};
+      double dist = ad_utility::detail::wktDistImpl(geo1, geo2) * 1000;
+      ASSERT_GT(static_cast<long long>(dist), maxDistInMeters);
+    }
+  };
+
+  auto testBounds = [](double x, double y, const box& bbox,
+                       bool shouldBeWithin) {
+    // correct lon bounds if necessary
+    if (x < -180) {
+      x += 360;
+    } else if (x > 180) {
+      x -= 360;
+    }
+
+    // testing only possible, if lat bounds are correct and the lon bounds
+    // don't cover everything (as then left or right of the box is again
+    // inside the box because of the spherical geometry)
+    double minLonBox = bbox.min_corner().get<0>();
+    double maxLonBox = bbox.max_corner().get<0>();
+    if (y < 90 && y > -90 && !(minLonBox < 179.9999 && maxLonBox > 179.9999)) {
+      bool within = boost::geometry::covered_by(point(x, y), bbox);
+      ASSERT_TRUE(within == shouldBeWithin);
+    }
+  };
+
+  // build dummy join to access the containedInBoundingBox and
+  // computeBoundingBox functions. Note that maxDistInMeters has to be accurate,
+  // otherwise the functions of spatialJoin don't work correctly
+  auto maxDistInMetersStr =
+      "<max-distance-in-meters:" + std::to_string(maxDistInMeters) + ">";
+  auto qec = buildTestQEC();
+  auto spatialJoinTriple =
+      SparqlTriple{TripleComponent{Variable{"?point1"}}, maxDistInMetersStr,
+                   TripleComponent{Variable{"?point2"}}};
+  std::shared_ptr<QueryExecutionTree> spatialJoinOperation =
+      ad_utility::makeExecutionTree<SpatialJoin>(qec, spatialJoinTriple,
+                                                 std::nullopt, std::nullopt);
+
+  // add children and test, that multiplicity is a dummy return before all
+  // children are added
+  std::shared_ptr<Operation> op = spatialJoinOperation->getRootOperation();
+  SpatialJoin* spatialJoin = static_cast<SpatialJoin*>(op.get());
+
+  PreparedSpatialJoinParams params{
+      nullptr,     nullptr, nullptr, nullptr,
+      0,           0,       1,       spatialJoin->getMaxDist(),
+      std::nullopt};
+
+  SpatialJoinAlgorithms spatialJoinAlgs{
+      qec, params, true, spatialJoin->onlyForTestingGetRealConfig()};
+
+  std::vector<box> bbox =
+      spatialJoinAlgs.OnlyForTestingWrapperComputeBoundingBox(startPoint);
+  // broad grid test
+  for (int lon = -180; lon < 180; lon += 20) {
+    for (int lat = -90; lat < 90; lat += 20) {
+      checkOutside(point(lon, lat), startPoint, bbox, &spatialJoinAlgs);
+    }
+  }
```

Comment:
Probably you should choose some distribution that has more values closer to the startPoint, because that's where it is interesting.

Reply:
i think that's not necessary as the testBounds function is doing exactly that: heavily testing the values barely in or out of the bounding box. This function checks everything but that so the combination covers everything.